### PR TITLE
src/stores: add behaviour to automatically refresh access token after login

### PR DESCRIPTION
**Pull request by maceksimon**
Token refresh behaviour in `login` store:

* Add API call to refresh token
* Add logic to read the token's expiration time (the `useJwt` composable)
* Add logic to run token refresh function on timeout

Token proactively refreshes so that we do not need to be refresh before making an API call.

---

TODO:
- [x] store timeout and remove after logout.

### CODE REVIEW

File: src/composables/useJwt.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        });
+      }
+      return null;
+    }
+  };
+
+  /**
+   * Parse JWT token
+   * @param token - JWT token
+   * @returns - JWT parts (header, payload, signature)
```
Comment (tmszi) [inline src/composables/useJwt.ts:48]: missing return type

File: src/composables/useJwt.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+  const decodePayload = (payload: string) => {
+    const jsonPayload = base64UrlDecode(payload);
+    return JSON.parse(jsonPayload);
+  };
+
+  /**
+   * Base64 URL decode
+   * @param str - Base64 URL encoded string
+   * @returns - Decoded string
```
Comment (tmszi) [inline src/composables/useJwt.ts:70]: Missing return type

File: src/composables/useJwt.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      throw new Error('Invalid JWT token format');
+    }
+    return {
+      header: parts[0],
+      payload: parts[1],
+      signature: parts[2],
+    };
+  };
+
+  const decodePayload = (payload: string) => {
```
Comment (tmszi) [inline src/composables/useJwt.ts:62]: Missing return type

File: src/composables/useJwt.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+          color: 'negative',
+        });
+      }
+      return null;
+    }
+  };
+
+  /**
+   * Parse JWT token
+   * @param token - JWT token
```
Comment (tmszi) [inline src/composables/useJwt.ts:47]: Missing token type

File: src/components/__tests__/FormLogin.cy.js [hunk 5]
```diff
@@ -22,19 +22,30 @@ const contactEmail = rideToWorkByBikeConfig.contactEmail;
...
+const { apiBase, urlApiLogin, urlApiRefresh } = rideToWorkByBikeConfig;
+const apiLoginUrl = `${apiBase}${urlApiLogin}`;
+const apiRefreshUrl = `${apiBase}${urlApiRefresh}`;
+
 const username = 'test@example.com';
 const password = 'example123';
-const tokenAccess = '1234567890';
-const tokenRefresh = '0987654321';
+
+const tokenAccess =
```
Comment (tmszi) [inline src/components/__tests__/FormLogin.cy.js:32]: token* var should be as Cy fixture.

File: src/composables/useJwt.ts (hunk 1 above)
Comment (maceksimon) [reply src/composables/useJwt.ts]: Updated docs.

File: src/composables/useJwt.ts [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+import { Notify } from 'quasar';
+
+// composables
+import { i18n } from '../boot/i18n';
+
+export const useJwt = () => {
+  /**
+   * Get JWT expiration
+   * @param token - JWT token
+   * @returns - Expiration time in seconds or null
```
[the remark this answers is not in the document]
Comment (maceksimon) [reply src/composables/useJwt.ts]: Updated docs.

File: src/composables/useJwt.ts (hunk 3 above)
Comment (maceksimon) [reply src/composables/useJwt.ts]: Updated docs and types.

File: src/composables/useJwt.ts (hunk 2 above)
Comment (maceksimon) [reply src/composables/useJwt.ts]: Updated docs.

File: src/composables/useJwt.ts (hunk 4 above)
Comment (maceksimon) [reply src/composables/useJwt.ts]: Updated docs.

File: src/components/__tests__/FormLogin.cy.js (hunk 5 above)
Comment (maceksimon) [reply src/components/__tests__/FormLogin.cy.js]: Refactored into a fixture.
Also updated the time variables in test for better readability.

File: ride_to_work_by_bike_config.toml [hunk 7]
```diff
@@ -72,3 +72,4 @@ apiBase = "https://test.dopracenakole.cz/rest/"
 apiVersion = "1.0"
 
 urlApiLogin = "auth/login/"
+urlApiRefresh = "auth/refresh/tokens/"
```
Comment (tmszi) [inline ride_to_work_by_bike_config.toml:75]: correct refresh token URL is auth/token/refresh/

File: src/composables/useJwt.ts [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+  /**
+   * Parse JWT token into parts
+   * @param {string} token - JWT token string
+   * @return {JwtParts} Object (header, payload, signature) - token parts
+   * @throws {Error} If the token format is invalid
+   */
+  const parseJwt = (token: string): JwtParts => {
+    const parts = token.split('.');
+    if (parts.length !== 3) {
+      throw new Error('Invalid JWT token format');
```
Comment (tmszi) [inline src/composables/useJwt.ts:61]: You can translate error message, please.

File: ride_to_work_by_bike_config.toml (hunk 7 above)
Comment (tmszi) [reply ride_to_work_by_bike_config.toml:75]: Because backend REST API Python lib was upgraded.

**Comment by tmszi** [comment]
```ts
interface LoginResponse {
  access_token: string;
  refresh_token: string;
  user: User;
```

should be

```ts
interface LoginResponse {
  access: string;
  refresh: string;
  user: User;
```

Because backend REST API Python lib was upgraded.

File: src/stores/login.ts [hunk 9]
```diff
@@ -31,6 +40,11 @@ interface User {
   last_name: string;
 }
 
+interface RefreshTokenResponse {
+  access: string;
+  access_token_expiration: string;
```
Comment (tmszi) [inline src/stores/login.ts:45]: `access_token_expiration` key should be ` access_expiration`.

File: ride_to_work_by_bike_config.toml (hunk 7 above)
Comment (maceksimon) [reply ride_to_work_by_bike_config.toml:75]: Updated endpoint.

File: src/composables/useJwt.ts (hunk 8 above)
Comment (maceksimon) [reply src/composables/useJwt.ts:61]: Translated.

File: src/stores/login.ts (hunk 9 above)
Comment (maceksimon) [reply src/stores/login.ts:45]: Updated.

**Comment by maceksimon** [comment]
> ```ts
> interface LoginResponse {
>   access_token: string;
>   refresh_token: string;
>   user: User;
> ```
> 
> should be
> 
> ```ts
> interface LoginResponse {
>   access: string;
>   refresh: string;
>   user: User;
> ```
> 
> Because backend REST API Python lib was upgraded.

Updated

**Review by tmszi — APPROVED** [review]

**Pull request merged**